library/core/src/intrinsics/mir.rs

```diff
@@ -249,6 +249,29 @@
 //!  - [`Call`] has an associated function as well. The third argument of this function is a normal
 //!    function call expression, for example `my_other_function(a, 5)`.
 //!
+//! #### Debuginfo
+//!
+//!  - A debuginfo name can be given to a local using `debug my_name => contents;`.
+//!    For `contents`, we use the same syntax as operands, to support both places and constants.
+//!
+//! ```rust
+//! #![allow(internal_features)]
+//! #![feature(core_intrinsics, custom_mir)]
+//!
+//! use core::intrinsics::mir::*;
+//!
+//! #[custom_mir(dialect = "built")]
+//! fn debuginfo(option: Option<&i32>) {
+//!     mir!(
+//!         debug option => option;
+//!         debug projection => *Field::<&i32>(Variant(option, 1), 0);
```

Comment:
So what does this mean? There's no local "projection" so I'm confused by both the left-hand side and the right-hand side.

Reply:
This mean debuginfo name `projection` corresponds to the place `(*((_1 as Some).0: &i32))`.

Reply:
And what does that mean?^^

If the intention is that this is only understandable for people that know our debuginfo system internals, then I guess it's fine. But as someone who has no idea at all about debuginfo, your explanation tells me nothing.

Reply:
I don't know the better terminology.
I by "debuginfo name", I mean the string you give to gdb when you want to print the variable.
MIR roughly represents debuginfo as a list of structs `{ Symbol, SourceScope, Place or Const }`, stating "this name at at this point in code corresponds to this memory place".

Reply:
Ah so it's meant to be the name of the variable in the original source code?

Reply:
Yes.
Should we call it "Source code variable name"?

Reply:
Yes I like that.